tensorflow_probability/python/experimental/mcmc/particle_filter.py

```diff
@@ -514,30 +660,42 @@ def _compute_observation_log_weights(step,
         ps.equal(step % num_transitions_per_observation, 0))
     observation_idx = step // num_transitions_per_observation
     observation = tf.nest.map_structure(
-        lambda x, step=step: tf.gather(x, observation_idx), observations)
+        lambda x, step=step: tf.gather(x,
+                                       observation_idx,
+                                       axis=particles_dim),
```

Comment:
I think this line should be unchanged -- because the shape of `observations` is not affected by particles_dim.